src/stream/src/executor/filter.rs

```diff
@@ -234,15 +234,16 @@ mod tests {
         };
         let source = MockSource::with_chunks(schema, PkIndices::new(), vec![chunk1, chunk2]);
 
-        let left_expr = InputRefExpression::new(DataType::Int64, 0);
-        let right_expr = InputRefExpression::new(DataType::Int64, 1);
-        let test_expr = new_binary_expr(
-            Type::GreaterThan,
+        let test_expr = build(
```

Comment:
I guess this PR can also enable us for https://github.com/risingwavelabs/risingwave/issues/7881? 🚀

Reply:
Not exactly this PR did, but the ideas behind them are similar. Based on my experience with egg, I feel that a pretty construction of expression is super useful and not hard to implement. I'll have a try on it.